lib/iris/_data_manager.py

```diff
@@ -5,37 +5,56 @@
 """Management of common state and behaviour for cube and coordinate data."""
 
 import copy
+import warnings
 
 import numpy as np
 import numpy.ma as ma
 
 from iris._lazy_data import as_concrete_data, as_lazy_data, is_lazy_data
+import iris.exceptions
+import iris.warnings
 
 
 class DataManager:
     """Provides a well defined API for management of real or lazy data."""
 
-    def __init__(self, data):
+    def __init__(self, data, shape=None):
         """Create a data manager for the specified data.
 
         Parameters
         ----------
-        data :
+        data : ArrayLike, optional
             The :class:`~numpy.ndarray` or :class:`~numpy.ma.core.MaskedArray`
             real data, or :class:`~dask.array.core.Array` lazy data to be
-            managed.
+            managed. If a value of ``None`` is given, the data manager will be
+            considered dataless.
+
+        shape : tuple, optional
+            A tuple, representing the shape of the data manager. This can only
+            be used in the case of ``data=None``, and will render the data manager
+            dataless.
 
         """
+        if (shape is None) and (data is None):
+            msg = 'one of "shape" or "data" should be provided; both are None'
+            raise ValueError(msg)
+        elif (shape is not None) and (data is not None):
+            if shape == data.shape:
```

Comment:
@ESadek-MO This breaks with `iris.cube.Cube(1, shape=(1,))` i.e., `data` may not be an instance with a shape method.

TBH thinking about this, doing a` data.shape` here is kinda dangerous. To avoid complications, it probably better to dial this back to ensuring that simply both `data` and `shape` are not provided.